include/alpaka/core/Cuda.hpp

```diff
@@ -50,151 +50,10 @@
-    #define ALPAKA_CUDA_RT_CHECK_IGNORE(cmd, ...)\
-        ::alpaka::cuda::detail::cudaRtCheckLastError("'" #cmd "' A previous CUDA call (not this one) set the error ", __FILE__, __LINE__);\
-        ::alpaka::cuda::detail::cudaRtCheckIgnore(cmd, #cmd, __FILE__, __LINE__, ##__VA_ARGS__)
-    #if BOOST_COMP_CLANG
-        #pragma clang diagnostic pop
-    #endif
-#endif
-
-//-----------------------------------------------------------------------------
-//! CUDA runtime error checking with log and exception.
-#define ALPAKA_CUDA_RT_CHECK(cmd)\
-    ALPAKA_CUDA_RT_CHECK_IGNORE(cmd)
-
-namespace alpaka
-{
-    namespace cuda
-    {
-        namespace detail
-        {
-            //-----------------------------------------------------------------------------
-            //! CUDA driver API error checking with log and exception, ignoring specific error values
-            ALPAKA_FN_HOST inline auto cudaDrvCheck(
-                CUresult const & error,
-                char const * desc,
-                char const * file,
-                int const & line)
-            -> void
-            {
-                if(error == CUDA_SUCCESS)
-                    return;
-
-                char const * cu_err_name = nullptr;
-                char const * cu_err_string = nullptr;
-                CUresult cu_result_name = cuGetErrorName(error, &cu_err_name);
-                CUresult cu_result_string = cuGetErrorString(error, &cu_err_string);
-                std::string sError = std::string(file)
-                                   + "(" + std::to_string(line) + ") "
-                                   + std::string(desc) + " : '";
-                if( cu_result_name == CUDA_SUCCESS && cu_result_string == CUDA_SUCCESS )
-                {
-                    sError += std::string(cu_err_name) +  "': '"
-                            + std::string(cu_err_string) + "'!";
-                } else {
-                    // cuGetError*() failed, so append corresponding error message
-                    if( cu_result_name == CUDA_ERROR_INVALID_VALUE ) {
-                        sError += " cuGetErrorName: 'Invalid Value'!";
-                    }
-                    if( cu_result_string == CUDA_ERROR_INVALID_VALUE ) {
-                        sError += " cuGetErrorString: 'Invalid Value'!";
-                    }
-                }
-#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
-                std::cerr << sError << std::endl;
-#endif
-                ALPAKA_DEBUG_BREAK;
-                throw std::runtime_error(sError);
-            }
-        }
-    }
-}
-
-//-----------------------------------------------------------------------------
-//! CUDA driver error checking with log and exception.
-#define ALPAKA_CUDA_DRV_CHECK(cmd)\
-    ::alpaka::cuda::detail::cudaDrvCheck(cmd, #cmd, __FILE__, __LINE__)
-
+#define ALPAKA_PP_CONCAT_DO(X,Y) X##Y
+#define ALPAKA_PP_CONCAT(X,Y) ALPAKA_PP_CONCAT_DO(X,Y)
```

Comment:
These two are general, maybe move to some alpaka preprocessor file (not sure if it exists), also to avoid duplication with HIP.

Reply:
I also thought about it but it is only needed here. @BenjaminW3 do you have a suggestion if we should stay with the two line duplication ore move it into an own file.

Reply:
If it is only used here, we can keep it here for now.